backend/src/libs/plugins/file-upload/file-upload.plugin.ts

```diff
@@ -0,0 +1,40 @@
+import { type MultipartFile } from '@fastify/multipart';
+import fp from 'fastify-plugin';
+
+import { ExceptionMessage } from '#libs/enums/enums.js';
+import { FileError } from '#libs/exceptions/exceptions.js';
+import { ControllerHook } from '#libs/packages/controller/controller.js';
+
+type Options = {
+  extensions: string[];
+};
+
+const fileUpload = fp<Options>((fastify, { extensions }, done) => {
+  fastify.decorateRequest('fileBuff', null);
+
+  fastify.addHook(ControllerHook.PRE_VALIDATION, async (request) => {
+    if (!request.isMultipart()) {
+      return;
+    }
+
+    const { file } = request.body as { file: MultipartFile };
+
+    if (file.file.truncated) {
+      return;
+    }
```

Comment:
I suggest we throw a file size error here and set ```throwFileSizeLimit``` to ```false``` in base-server-application.ts
The reason is that with ```throwFileSizeLimit``` set to ```true``` the library will throw an error message, which is not user-friendly.